src/json_decode.ml

```diff
@@ -95,6 +95,13 @@ let field key decode json =
   else
     raise @@ Decode_error ("Expected object, got " ^ Js.Json.stringify json)
 
+let at topKey key_path decoder =
+  let rec at l decoder prev =
+    match l with 
+      | x::t -> at t (field x decoder) prev
+      | [] -> field prev decoder in
+    at (List.rev key_path) decoder topKey 
```

Comment:
There are several issues here:

1. This can be implemented without needing to "pre-reverse" the list, see e.g. https://github.com/ocaml/ocaml/blob/trunk/stdlib/list.ml#L90-L95

2. I'm leaning towards not having an explicit `topKey` argument. Even if that would make it somewhat unsafe, it makes it more intuitive and simpler to both use and implement, and it's very obvious it would fail if passed an empty list anyway, so I think it's an acceptable compromise.

3. `x::t` isn't very consistent. It should either be `x::xs` or `h::t`, and preferably the same you use elsewhere